eisbuk-admin/src/components/atoms/AttendanceCard/AttendanceCard.tsx

```diff
@@ -36,28 +66,86 @@ const AttendanceCard: React.FC<Props> = ({
   const endTime = luxonEnd.toISOTime().substring(0, 5);
 
   const timeString = `${startTime} - ${endTime}`;
+
   return (
-    <div>
-      <div data-testid="time-string">{timeString}</div>
-      <div>{categories}</div>
-      {userBookings.map((user) => {
-        const isAbsent = absentees?.includes(user.customer_id) || false;
-        return (
-          <div key={user.customer_id}>
-            <div>{user.name}</div>
-            <button
-              type="button"
-              onClick={() =>
-                dispatch(markAttendance(user.customer_id, isAbsent))
+    <Container maxWidth="sm">
+      <List className={classes.root}>
+        <div key={id + "-" + timeString} className={classes.slotWrapper}>
+          <ListItem className={classes.listHeader}>
+            <ListItemText
+              primary={
+                <span>
+                  {timeString} <b>({userBookings.length})</b>
+                </span>
               }
-            >
-              {isAbsent ? "👎" : "👍"}
-            </button>
-          </div>
-        );
-      })}
-    </div>
+              secondary={translateAndJoinTags(categories, type)}
+            />
+          </ListItem>
+          {userBookings.map((user) => {
+            const isAbsent = absentees?.includes(user.customer_id) || false;
+            const listItemClass = isAbsent ? classes.absent : "";
+
+            const absenteeButtons = (
+              <Button
+                variant="contained"
+                size="small"
+                color={isAbsent ? "primary" : "secondary"}
+                onClick={() =>
+                  dispatch(markAttendance(user.customer_id, isAbsent))
+                }
+                // disabled={hasLocalChange}
+              >
+                {isAbsent ? "👎" : "👍"}
+              </Button>
+            );
+
+            return (
+              <ListItem
+                key={`${id}-${user.customer_id}`}
+                className={listItemClass}
+              >
+                <ListItemAvatar>
+                  <EisbukAvatar {...user} />
+                </ListItemAvatar>
+                <ListItemText primary={user.name} />
+                <ListItemSecondaryAction>
+                  {absenteeButtons}
+                </ListItemSecondaryAction>
+              </ListItem>
+            );
+          })}
```

Comment:
So yeah, this is ugly (having a large render mapping function inside return statement (a really hard to read).

Reply:
The solution in my opinion would be to just create a `UserAttendance` component and move all of the rendering logic (and onClick events and all) there and only pass the necessary props.
This would greatly shorten the return of the mapping function and result in much more readable code 
```typescript
// note: this is just an example
{userBookings.map((userBooking) => <UserAttendance {...appropriateProps} />)}
```